src/strategies/saffron-finance/index.ts

```diff
@@ -0,0 +1,268 @@
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+) {
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+
+  // ================ Dex LP Token Reserve and Total Supply ==================
+  let dexReserveData = new Array();
+  let dexCalls = new Array();
+  let dexQueryIdx = 0;
+  options.dexLpTypes.forEach(dexToken => {
+    let d = {
+      name: dexToken.name,
+      reservesQuery: [dexToken.lpToken, 'getReserves'],
+      reserveQueryIdx: 0,
+      reserve: 0,
+      supplyQuery: [dexToken.lpToken, 'totalSupply'],
+      supplyQueryIdx: 0,
+      supply: 0,
+      saffLpToSFI_E18: 0
+    }
+    dexCalls.push(d.reservesQuery);
+    d.reserveQueryIdx = dexQueryIdx++;
+    dexCalls.push(d.supplyQuery);
+    d.supplyQueryIdx = dexQueryIdx++;
+    dexReserveData.push(d);
+  });
+
+  const dexResponse = await multicall(
+    network,
+    provider,
+    abi,
+    [
+      ...dexCalls
+    ],
+    {blockTag}
+  );
+
+  dexReserveData.forEach(drd => {
+    drd.reserve = dexResponse[drd.reserveQueryIdx][0];
+    drd.supply = dexResponse[drd.supplyQueryIdx][0];
+    drd.saffLpToSFI_E18 = drd.reserve.mul(BIG18).div(drd.supply);
+  });
+
+
+  // ================== One to One Voting Power ==================
+  let oneToOneData = new Array();
+  for (let i = 0; i < options.oneToOne.length; i++) {
+    let oto = options.oneToOne[i];
+    let otoRecord = {
+      label: oto.label,
+      tokenAddress: oto.tokenAddress,
+      otoQuery: addresses.map((address: any) => [
+        oto.tokenAddress,
+        'balanceOf',
+        [address]
+      ]),
+      otoResponses: new Array(),
+      votingScores: new Array()
+    };
+    otoRecord.otoResponses = await multicall(network, provider, abi, otoRecord.otoQuery, {blockTag});
+    otoRecord.votingScores = addresses.map((address: any, index: number) => {
+      return {address: address, score: otoRecord.otoResponses[index][0]}
+    });
+    oneToOneData.push(otoRecord);
+  }
+
+  // =============== Boosted with Multiplier Voting Power ================
+  let boostMultiData = new Array();
+  for (let i = 0; i < options.boostMultiply.length; i++) {
+    let boost = options.boostMultiply[i];
+    let boostRecord = {
+      vote: boost.vote,
+      tokens: new Array()
+    };
+    const voteBoost1000 = BigNumber.from(boostRecord.vote * 1000);
+    const voteBoostDiv = BigNumber.from(1000);
+
+    for (let j = 0; j < options.boostMultiply[i].tokens.length; j++) {
+      let token = options.boostMultiply[i].tokens[j];
+      let tokenRecord = {
+        label: token.label,
+        queries: new Array(),
+        boostResponses: new Array(),
+        votingScores: new Array()
+      }
+      let queries = addresses.map((address: any) => [
+        token.tokenAddress,
+        "balanceOf",
+        [address]
+      ]);
+      tokenRecord.queries.push(queries);
+      let response = await multicall(network, provider, abi, queries, {blockTag});
```

Comment:
The way the strategy is designed is not going to scale well, the result of a strategy is calculated on user browser i recommend to send a maximum of 3 multicall request in a single strategy but this strategy seem to be sending more than 30 multicall.

Reply:
Okay, thank you. I can create an implementation with that restriction.